packages/app/src/search/searchers/contact.searcher.test.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import elasticlunr from "elasticlunr"
+import { AppError } from "App/core/errors"
+import { Contact } from "App/contacts/dto"
+import { SearcherError } from "App/search/constants"
+import { IndexFactory } from "App/index-storage/factories"
+import { DataIndex } from "App/index-storage/constants"
+import { ContactSearcher } from "App/search/searchers/contact.searcher"
+
+const indexContactMock = elasticlunr<Contact>()
+const contactMock: Contact = {
+  id: "0",
+  firstName: "Sławomir",
+  lastName: "Borewicz",
+  primaryPhoneNumber: "+71195069214",
+  secondaryPhoneNumber: "",
+  email: "example@mudita.com",
+  note: "sapiente rem dignissimos sunt",
+  ice: false,
+  favourite: false,
+  blocked: false,
+  firstAddressLine: "Malczewskiego 3, Warszawa",
+  secondAddressLine: "",
+}
+
+describe("When index exists", () => {
+  const index = new IndexFactory().create()
+  const subject = new ContactSearcher(index)
+
+  beforeAll(() => {
+    index.set(DataIndex.Contact, indexContactMock)
+
+    indexContactMock.setRef("id")
+    indexContactMock.addField("firstName")
+    indexContactMock.addField("lastName")
+    indexContactMock.addField("primaryPhoneNumber")
+    indexContactMock.addField("secondaryPhoneNumber")
+    indexContactMock.addField("email")
+    indexContactMock.addField("note")
+    indexContactMock.addField("ice")
+    indexContactMock.addField("favourite")
+    indexContactMock.addField("blocked")
+    indexContactMock.addField("firstAddressLine")
+    indexContactMock.addField("secondAddressLine")
+
+    indexContactMock.addDoc(contactMock)
+  })
+
+  test("returns hydrated `contact` list if query match to string in fields", () => {
+    expect(subject.search("Malczewskiego")).toEqual([contactMock])
+  })
+
+  test("returns empty array if query doesn't match to string in fields", () => {
+    expect(subject.search("Hello")).toEqual([])
+  })
+})
+
```

Comment:
https://github.com/mudita/mudita-center/pull/935/commits/b5b393fbf2f2b1cb7ed31ef20f1f347617866c77

Reply:
@igorbezsmertnyi Do we want to change the search logic for the contacts view? Or we would like to keep this logic for a feature in case it happens?

Reply:
@dkarski Nothing

Reply:
so this is a dead code? 🤔 What is the benefit of that?

Reply:
I will provide you with an example. What will happen if you try to search the contacts in another view, where you have a direct connection to `contacts`? So, to prevent logic duplication and stay consistent, it's better to have this part of the code instead of not have.

Reply:
Good thing we have a generic solution for that, like a search module. This is great. However (and this is just my opinion, not a rule), I think that we can avoid a particular implementation when the implementation isn't used, to avoid massless of dead code.